extensions/cornerstone-dynamic-volume/src/getSopClassHandlerModule.ts

```diff
@@ -0,0 +1,340 @@
+    volumesTimePointsCache.set(referencedVolume, timePointsData);
+  }
+
+  const { timePoints, timePointsUnit } = timePointsData;
+
+  if (timePoints.length !== segPixelDataInTime[0].length) {
+    throw new Error('Invalid number of time points returned');
+  }
+
+  const timepointsCount = timePoints.length;
+  const chartSeriesData = new Array(timepointsCount);
+
+  for (let i = 0; i < timepointsCount; i++) {
+    const average = segPixelDataInTime.reduce(
+      (acc, cur) => acc + cur[i] / pixelCount,
+      0
+    );
+
+    chartSeriesData[i] = [timePoints[i], average];
+  }
+
+  return {
+    StudyInstanceUID,
+    StudyDescription,
+    chartData: {
+      series: {
+        label: segmentation.label,
+        points: chartSeriesData,
+      },
+      axis: {
+        x: {
+          label: `Time (${timePointsUnit})`,
+        },
+        y: {
+          label: `Vl (Bq/ml)`,
+        },
+      },
+    },
+  };
+}
+
+function _getInstanceFromSegmentations(segmentations) {
+  if (!segmentations.length) {
+    return;
+  }
+
+  const volumesTimePointsCache = new WeakMap();
+  const segmentationsData = segmentations.map(segmentation =>
+    _getSegmentationData(segmentation, volumesTimePointsCache)
+  );
+
+  const { date: seriesDate, time: seriesTime } = _getDateTimeStr();
+  const seriesInstanceUid = _getNewSeriesInstanceUid(seriesDate, seriesTime);
+
+  const instance = {
+    SOPClassUID: ChartDataSOPClassUid,
+    Modality: CHART_MODALITY,
+    SeriesDate: seriesDate,
+    SeriesTime: seriesTime,
+    SeriesInstanceUID: seriesInstanceUid,
+    StudyInstanceUID: segmentationsData[0].StudyInstanceUID,
+    StudyDescription: segmentationsData[0].StudyDescription,
+    SeriesNumber: 100,
+    SeriesDescription: 'Segmentation chart series data',
+    chartData: {
+      series: segmentationsData.reduce(
+        (allSeries, curSegData) => [...allSeries, curSegData.chartData.series],
+        []
+      ),
+      axis: { ...segmentationsData[0].chartData.axis },
+    },
+  };
+
+  const seriesMetadata = {
+    StudyInstanceUID: instance.StudyInstanceUID,
+    StudyDescription: instance.StudyDescription,
+    SeriesInstanceUID: instance.SeriesInstanceUID,
+    SeriesDescription: instance.SeriesDescription,
+    SeriesNumber: instance.SeriesNumber,
+    SeriesTime: instance.SeriesTime,
+    SOPClassUID: instance.SOPClassUID,
+    Modality: instance.Modality,
+  };
+
+  return { seriesMetadata, instance };
+}
+
+function _updateSegmentationsDisplaySets(appContext) {
+  const { servicesManager } = appContext;
+  const { segmentationService } = servicesManager.services;
+
+  const segmentations = segmentationService.getSegmentations();
+  const { seriesMetadata, instance } = _getInstanceFromSegmentations(
+    segmentations
+  );
+
+  // An event is triggered after adding the instance and the displaySet is created
+  DicomMetadataStore.addSeriesMetadata([seriesMetadata], true);
+  DicomMetadataStore.addInstances([instance], true);
+}
```

Comment:
I'm surprised this exists here, why the sopHandler for the line chart should care about segmentation?